app/models/speaker.rb

```diff
@@ -9,8 +9,15 @@ class Speaker < ActiveRecord::Base
 
   serialize :info, Hash
 
-  validates :user, :event, presence: true
+  validates :event, presence: true
   validates :bio, length: {maximum: 500}
+  validates :name, :email, presence: true
+
+  scope :with_program_session, -> { Speaker.where("program_session_id IS NOT NULL") }
+
+  scope :for_event, -> (current_event) do
```

Comment:
Do you mean something like:
<code>has_many :public_session_formats, ->{ Speaker.where("program_session_id IS NOT NULL")) }, class_name: Speaker</code>
in the Event model? I haven't done a has_many like that before, so I'm a little unclear on this.